### DISCUSSION TITLE
Deleting an ephemeral message

### DISCUSSION THREAD

**Discussion by Nekonika** [discord-net/Discord.Net#2705] [Q&A] [answered]
Greetings everyone :)

I've asked this in your Discord-Server already, but either the answer to the question is way to easy and they do not event want to answer it, or the people who read it did not know the answer either....
So anyways: I've got an Embed set up as kind of like a ticket dashboard and added a few buttons beneath, where one of for example would be "Create Whitelist Ticket".
The Idea is, that if the User, who clicked the button is already whitelisted, he gets an ephemeral response within that channel saying that he is already whitelisted. I want this message to be deleted after lets say 5 seconds or so...

This is where I get to my actual question: Discords API Documentation says, that it is possible to delete an ephemeral message, but you would have to use another endpoint which requires the interaction token in order to delete the message.
How would I approach that?

This is my code so far:
```cs
[ComponentInteraction("create_ticket_whitelist")]
public async Task CreateWhitelistTicket()
{
    await DeferAsync(ephemeral: true);
    // doing some work here
    
    await FollowupAsync(embed: GetEmbed("CREATEWLTICKET"), ephemeral: true);
    _ = Task.Delay(5 * 1000) // wait for 5 seconds
        .ContinueWith(async _ => await Context.Interaction.DeleteOriginalResponseAsync()) // delete the ephemeral message
        .ConfigureAwait(false);
}
```

This piece of code though deletes the dashboard message instead of the ephemeral response.
Using the `IUserMessage` returned by `FollowupAsync()` wouldn't work, as (afaik) it does not store the Interaction Token which would be required to use the API endpoint which allows deleting ephemeral messages.
I can get around this when using `RespondAsync()` instead of `FollowupAsync()` and removing the `DeferAsync()`...
This works just fine, but then I would run into issues if my `// doing some work here` takes longer than 3 seconds, as after 3s I can no longer respond to an Interaction.

I could provably use the information provided by the Library and call the endpoint myself, but I guess that would destroy the whole purpose of using the library at all...
I would greatly appreciate any help whatsoever, as I just cant find the right way to do it myself...

Thank you guys for reading
and thank you in advance for any help whatsoever :D

Kindly,
Nekonika

**Comment by Misha-133** [comment, marked as answer]
Well
Using the `IUserMessage` returned by `FollowupAsync()` is actually the right way here.
You just need to cast it to `RestFollowupMessage` & call `DeleteAsync()` on it. It will make a call to `Delete Followup Message` endpoint and delete the msg.

*just a small example*
```cs
var msg = await FollowupAsync(...) as RestFollowupMessage;
_ = Task.Delay(TimeSpan.FromSeconds(5)).ContinueWith(_ =>msg.DeleteAsync()));
```

**Reply by Nekonika** [reply]
Great answer!
I'll give it a try as soon as im back home from work.

While we're at it... Is there a way I could have figured this out myself?
Is this documented anywhere, or will I have to dig into the libraries sourcecode to find such information?
I just want to make sure, that I dont need to ask any silly question like this again :D

**Reply by Misha-133** [reply]
Well... That's the thing with DNet
A lot of stuff is not properly documented (aside from XMLdoc), and since quite a lot of features are hidden behind casts like this - it gets extra confusing.
*Hopefully this will change in v4*

So ye, in most cases you'd need to look at inheritance trees in docs... Or dig through source code

*Or search through `#help` forum in the discord server*

**Reply by Nekonika** [reply]
I just tried what you said and everything seems to be working just fine.
Thank you for your help :D
